.golangci-kal.yml

```diff
@@ -39,13 +39,13 @@ linters-settings:
             isFirstField: Warn # Require conditions to be the first field in the status struct.
             usePatchStrategy: Forbid # Conditions should not use the patch strategy on CRDs.
             useProtobuf: Forbid # We don't use protobuf, so protobuf tags are not required.
-        # jsonTags:
-        #   jsonTagRegex: "^[a-z][a-z0-9]*(?:[A-Z][a-z0-9]*)*$" # The default regex is appropriate for our use case.
-        # optionalOrRequired:
-        #   preferredOptionalMarker: optional | kubebuilder:validation:Optional # The preferred optional marker to use, fixes will suggest to use this marker. Defaults to `optional`.
-        #   preferredRequiredMarker: required | kubebuilder:validation:Required # The preferred required marker to use, fixes will suggest to use this marker. Defaults to `required`.
-        # requiredFields:
-        #   pointerPolicy: Warn | SuggestFix # Defaults to `SuggestFix`. We want our required fields to not be pointers.
+          jsonTags:
+            jsonTagRegex: "^[a-z][a-z0-9]*(?:[A-Z][a-z0-9]*)*$" # The default regex is appropriate for our use case.
```

Comment:
Do we want to uncomment this, or shall we just leave it commented so that if the default configuration changes over time we continue to use the default?

Reply:
The CI will be failed if KAL changes default configuration. Then we'll end up adding the this configuration to manage the compability.
So I think it's better to keep this configuration.

Reply:
CI will only fail if we bump KAL on the PR where we bump KAL

Reply:
@JoelSpeed How do we keep the documented default values here up-to-date if they change in KAL?

Reply:
(In general I'm in favor of using the default value here from KAL)

Reply:
> @JoelSpeed How do we keep the documented default values here up-to-date if they change in KAL?

Open to suggestions, I don't know at the moment.

I'm expecting in the not too distant future to move the project to kube-sigs. Once it's moved, and we've had agreement from sig-apimachinery that we are happy with the structure and defaults etc, that we probably will start treating the project like a v1 and therefore breaking changes won't be possible.

So I guess, as we update kal, we add the new sections in to show what the new config might be, but I don't have a great/automatic suggestion at the moment

Reply:
All good. We can just sync the defaults once we reach "v1"

Reply:
Agreed. thx!